### PR TITLE
Avoid a few unnecessary char.ToStrings

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ internal static void GetDefaultIdentityAndHash(out object identity, out string h
                 hash = IdentityHelper.GetNormalizedStrongNameHash(assemblyName)!;
                 if (hash != null)
                 {
-                    hash = "StrongName" + separator + hash;
+                    hash = string.Concat("StrongName", new ReadOnlySpan<char>(in separator), hash);
```
Comment (EgorBo) [inline src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs:62]: Is this something Roslyn could do?

File: src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs [hunk 2]
```diff
@@ -109,7 +109,10 @@ internal static void ExtractRelativeToDirectory(this ZipArchiveEntry source, str
             DirectoryInfo di = Directory.CreateDirectory(destinationDirectoryName);
             string destinationDirectoryFullPath = di.FullName;
             if (!destinationDirectoryFullPath.EndsWith(Path.DirectorySeparatorChar))
-                destinationDirectoryFullPath += Path.DirectorySeparatorChar;
+            {
+                char sep = Path.DirectorySeparatorChar;
```
Comment (EgorBo) [inline src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs:113]: can it be replaced with `PathInternal.DirectorySeparatorCharAsString` too?

File: src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs:62]: Will need to think about this one. But for the const, come to think of it I thought I fixed that case in Roslyn years ago... will look into it...

Comment (stephentoub) [reply src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs:62]: Ah, I had it in my head that Path.DirectorySeparatorChar was a public const, but it's actually a public static readonly. The change I made to Roslyn years ago was about promoting const chars to const strings, and that's still in effect.

File: src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs:113]: It _could_.  However, that's defined in a pair of files (PathInternal.Windows.cs / PathInternal.Unix.cs) that aren't currently included in this project's build.  I could pull them in as well, as this project does build once for Windows and once for Unix, but doing so then also requires pulling in other files, like the source for ValueStringBuilder, and it didn't seem worth it.

**Review by EgorBo — APPROVED** [review]

File: src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs:62]: I opened https://github.com/dotnet/roslyn/issues/66827.

**Pull request merged**